CleanWeatherTests/Scenes/CityForecast/CityForecastInteractorTests.swift

```diff
@@ -6,12 +6,90 @@
 //  Copyright © 2019 vandermesis. All rights reserved.
 //
 
-import XCTest
 import Quick
 import Nimble
 
 @testable import CleanWeather
 
-class CityForecastInteractorTests: XCTestCase {
+final class CityForecastInteractorTests: QuickSpec {
 
+    override func spec() {
+
+        var presenter: FakeCityForecastPresenter!
+        var worker: FakeCityForecastWorker!
+        var router: FakeCityForecastRouter!
+        var interactor: CityForecastInteractorImpl!
+
+        beforeEach {
+            presenter = FakeCityForecastPresenter()
+            worker = FakeCityForecastWorker()
+            router = FakeCityForecastRouter()
+            interactor = CityForecastInteractorImpl(cityDetails: Mock.cityWeather2, presenter: presenter, worker: worker, router: router)
+        }
+
+        describe("getting city details") {
+
+            beforeEach {
+                interactor.getCityDetails()
+            }
+
+            it("should call presenter to display valid city details") {
+                expect(presenter.displayCityDetailsCalled?.city).to(equal(Mock.cityWeather2.city))
+            }
+        }
+
+        describe("getting city forecast") {
+
+            beforeEach {
+                interactor.getCityForecast()
+            }
+
+            it("should call presenter to show spinner") {
+                expect(presenter.toggleSpinnerStateCalled).to(beTrue())
+            }
+
+            it("should call worker to fetch city hourly details list for valid id") {
+                expect(worker.fetchCityHourDetailsListCalled).to(beTrue())
+                expect(worker.fetchCityHourDetailsListIdCalled).to(equal(Mock.cityWeather2.id))
+            }
+
+            context("on success response") {
+
+                beforeEach {
+                    worker.fetchCityHourDetailsListCompletion?(.success(Mock.cityForecast))
+                }
+
+                it("should call presenter to hide spinner") {
+                    expect(presenter.toggleSpinnerStateCalled).to(beFalse())
+                }
+
+                it("should call presenter to display valid hourly city details list for valid id") {
+                    expect(presenter.displayCityDetailsListCalled).notTo(beNil())
+                    expect(presenter.displayCityDetailsListCalled?.count).to(equal(Mock.cityForecast.count))
+                    expect(Mock.cityWeather2.id).to(equal(Mock.cityForecast1.id))
```

Comment:
this never changes, youre testing 2 mocked values, that makes no sense and is not related with fetching city hour details list youre now testing